### ISSUE TITLE
[Android] last git version won't install or compile

### ISSUE THREAD

**Issue by rborn** [tanguyantoine/react-native-music-control#332]
# Description

1. Sample code (provide repo url or sample code)
Hi everyone, 
I'm trying to use the git version but I'm hitting a wall.

A simple `npm install git://github.com/tanguyantoine/react-native-music-control.git` will give me some tsc errors

~~~
(letsemjoyapp)--(bugfix/musiccontrolupdate*)==>
$ npm install git://github.com/tanguyantoine/react-native-music-control.git

> react-native-music-control@0.11.0 prepack /tmp/bundle/vendor/.npm/_cacache/tmp/git-clone-6e78f400
> tsc -p tsconfig.json

src/constants.android.ts:1:31 - error TS2307: Cannot find module 'react-native' or its corresponding type declarations.

1 import { NativeModules } from 'react-native'
                                ~~~~~~~~~~~~~~
~~~

And if  I try to install ignoring the scripts, it won't compile


~~~
/Work/clients/current/emjoy/app/letsemjoyapp/node_modules/react-native-music-control/android/src/main/java/com/tanguyantoine/react/MusicControlNotification.java:189: error: cannot find symbol
            notification = MusicControlModule.INSTANCE.notification.prepareNotification(MusicControlModule.INSTANCE.nb, false);
            ^
  symbol:   variable notification
  location: class NotificationService
/Work/clients/current/emjoy/app/letsemjoyapp/node_modules/react-native-music-control/android/src/main/java/com/tanguyantoine/react/MusicControlNotification.java:190: error: cannot find symbol
            startForeground(NOTIFICATION_ID, notification);
                                             ^
  symbol:   variable notification
  location: class NotificationService
/Work/clients/current/emjoy/app/letsemjoyapp/node_modules/react-native-music-control/android/src/main/java/com/tanguyantoine/react/MusicControlNotification.java:191: error: cannot find symbol
            isRunning = true;
            ^
  symbol:   variable isRunning
  location: class NotificationService
Note: /Work/clients/current/emjoy/app/letsemjoyapp/node_modules/react-native-music-control/android/src/main/java/com/tanguyantoine/react/MusicControlAudioFocusListener.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
3 errors

~~~





<!-- replace the balnk space within backets by a x so that it checks the box ;-) -->
2. Platform ?

   - [ ] iOS
   - [X] Android
   
<!-- same here -->   
3. Device
  - [ ] Simulator
  - [X] Real device


Thank you!

**Comment by bradleyflood** [comment]
Hi @rborn this is likely due to the publishing process. You'll need to install via NPM as per the instructions. But if you can't do that then you could copy the NPM folder and upload it to your own fork or something.
Cheers
Brad

**Comment by rborn** [comment]
Hi @bradleyflood, thanks for replying. I am trying to fix it in my fork but please reopen the issue as it doesn't compile and there are errors in the java code (on it, will open a PR as soon as I have it solved)

**Comment by bradleyflood** [comment]
Hi @rborn take a look at this commit here in my dev fork.
https://github.com/bradleyflood/react-native-music-control/commit/713208ba3da40204bb19a7d541e1ede255515632

These are the files that are generated when running the build commands. These are the files you are missing.
At the very bottom of the above commit you will see the package.json build commands. You can either run those or just copy the files from the commit above.

Why not just use NPM?

**Comment by rborn** [comment]
The missing files are for typescript, which is fine, I can leave without them :) but there are errors in java and won't compile.

And the npm version is 4 months old, not the current one, is not yet published 😄  https://www.npmjs.com/package/react-native-music-control

**Comment by bradleyflood** [comment]
Haha ok awesome you understand this situation better than I do. Let's re-open it and look into it.

Sorry my bad!

**Comment by rborn** [comment]
@bradleyflood ok, PR done

It is a mix between other commits and this fix

- added missing libs so install from git works (typescript)
- uses a commit from somebody else that adds a check for abandonFocus
- fixes the undefined compile issue
- removes an unused isRunning var

I installed from my fork and compiled the app, all works :)

**Comment by bradleyflood** [comment]
Thanks @rborn merged and released in `0.12.0`

Good job mate!
